hstream/src/HStream/Server/Types.hs

```diff
@@ -128,11 +128,11 @@ data Assignment = Assignment
 data ConsumerWorkload = ConsumerWorkload
   { cwConsumerName :: ConsumerName,
     cwShardCount   :: Int
-  }
+  } deriving (Show)
 instance Eq ConsumerWorkload where
   (==) w1 w2 = cwConsumerName w1 == cwConsumerName w2 && cwShardCount w1 == cwShardCount w2
 instance Ord ConsumerWorkload where
-  (<=) w1 w2 = w1 == w2 || cwShardCount w1 <= cwShardCount w2
+  (<=) w1 w2 = (cwShardCount w1, cwConsumerName w1) <= (cwShardCount w2, cwConsumerName w2)
```

Comment:
Why not `deriving (Eq, Ord, Show)` ?

Reply:
@4eUeP ``deriving (Eq, Show)`` is ok, but for ``Ord``, we want to compare ``cwShardCount`` before ``cwConsumerName``, which is used to find the max ``cwShardCount`` of ``Set<ConsumerWorkload>``

Reply:
``ConsumerWorkload`` should be ordered by ``shardCount`` first.

Reply:
got, thks